internal/workload/manager.go

```diff
@@ -539,3 +544,29 @@ func (w *WorkloadManager) podAuthModified(authPath string, auth string) bool {
 	}
 	return !bytes.Equal(file, []byte(auth))
 }
+
+func (w *WorkloadManager) updateSecrets(configuration models.DeviceConfigurationMessage) []error {
```

Comment:
I need to say that here instead of models.DeviceConfigurationMessage, I would only send the secrets object instead of all config.